spec/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_spec.rb

```diff
@@ -1,11 +1,181 @@
+      config_script = manager.configuration_scripts.first
+      external      = {:some_key => :some_value}
+      internal      = config_script.variables
+      merged        = config_script.send(:merge_extra_vars, external)
+
+      expect(internal).to be_a Hash
+      expect(merged).to eq(:instance_ids => ["i-3434"], :some_key => :some_value)
+    end
+
+    it "merges an internal hash and an empty hash to send out to ansible_runner" do
+      config_script = manager.configuration_scripts.first
+      external      = nil
+      merged        = config_script.send(:merge_extra_vars, external)
+
+      expect(merged).to eq(:instance_ids => ["i-3434"])
+    end
+
+    it "merges an empty internal hash and a hash to send out to the tower gem" do
+      config_script = manager.configuration_scripts.first.tap { |cs| cs.variables = {} }
+      external      = {:some_key => :some_value}
+      merged        = config_script.send(:merge_extra_vars, external)
+
+      expect(merged).to eq(external)
+    end
+
+    it "merges all empty arguments to send out to the tower gem" do
+      config_script = manager.configuration_scripts.first.tap { |cs| cs.variables = {} }
+      external      = nil
+      merged        = config_script.send(:merge_extra_vars, external)
+
+      expect(merged).to eq({})
+    end
+
+    it "decrypts extra_vars before sending out to the tower gem" do
+      config_script = manager.configuration_scripts.first
+      password      = "password::#{ManageIQ::Password.encrypt("some_value")}"
+      external      = {:some_key => password}
+      merged        = config_script.send(:merge_extra_vars, external)
+
+      expect(merged).to eq(:instance_ids => ["i-3434"], :some_key => "some_value")
+    end
   end
 
-  it_behaves_like 'ansible configuration_script'
+  # was "CUD via the API"
+  context "CRUD operations" do
+    let(:manager) { FactoryBot.create(:embedded_automation_manager_ansible, :provider) }
+
+    let(:params) do
+      {
+        :description  => "Description",
+        :extra_vars   => {}.to_json,
+        :inventory_id => 1,
+        :playbook     => playbook.name,
+        :name         => "My Job Template",
+        :related      => {}
+      }
+    end
+
+    context ".create_in_provider" do
+      it "successfully created in provider" do
+        new_config_script = described_class.create_in_provider(manager.id, params)
+
+        expect(new_config_script).to be_a(described_class)
+        expect(new_config_script.manager_id).to eq(manager.id)
+      end
+
+      it "not found during refresh" do
+        params[:playbook] = "not_a_playbook.yaml"
+        error_msg         = 'Playbook name="not_a_playbook.yaml" no longer exists'
+
+        expect do
+          described_class.create_in_provider(manager.id, params)
+        end.to raise_error(RuntimeError, error_msg)
+      end
+
+      # TODO:  Determine if we want to have a uniqueness validation to
+      # replicate this functionality, otherwise delete this case.
```

Comment:
Also, forgot to bring this up (or maybe I did and it is buried in the review comments here), but curious your thoughts on this comment.  Might make sense, might not.

Reply:
I don't think we really have an analogous case because we're not creating an external reference for the job run. We can probably delete this bit.